src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -181,6 +181,14 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $opt
 							$this->nextAutoIndexes[] = $newAutoIndex;
 						}
 					}
+					if (count($this->keyTypes) === 1 && $offsetType->getValue() < 0) {
```

Comment:
Changes to the builder certainly need new tests in ConstantArrayTypeBuilderTest.